### ISSUE TITLE
Oracle RAC

### ISSUE THREAD

**Issue by lauracowen** [OpenLiberty/docs#3351]
Should info about [Oracle RAC](https://www.ibm.com/support/knowledgecenter/SSAW57_liberty/com.ibm.websphere.wlp.nd.multiplatform.doc/ae/twlp_oraclerac.html) be included in the [OL JDBC config topic](https://www.openliberty.io/docs/20.0.0.12/relational-database-connections-JDBC.html)? (ask SME)

Is it that straightforward or would you only use Oracle RAC for certain purposes? If so, consider what the target user needs to know.

SME contact: @aguibert

**Comment by dmuelle** [comment]
https://www.oracle.com/database/real-application-clusters/

**Comment by frowe** [comment]
Yes, RAC should be in the OL config topic.  RAC is a very popular HA solution for Oracle users.  It is typically used in conjunction with the UCP configuration.

**Comment by dmuelle** [comment]
Hi @frowe , when you have a chance can you review the Oracle RAC example on draft:

https://docs-draft-openlibertyio.mybluemix.net/docs/21.0.0.3/relational-database-connections-JDBC.html#Oraclerac

want to confirm that there are no maven coordinates associated with RAC like there are with the JDBC drivers and Oracle UCP. I looked and didn't find any.

**Comment by frowe** [comment]
Hello @dmuelle Looks good, a few comments.  This statement (which came from the KC topic) has a typo
"If you are not using Oracle services, then the value for SERVICE_NAME is your database name. If you are using Oracle services, then the value for SERVICE_NAME is the name of the services."
should be 
"If you are not using Oracle services, then the value for SERVICE_NAME is your database name. If you are using Oracle services, then the value for SERVICE_NAME is the name of the service." (no ess)
This statement (also from KC topic) seems a bit awkward
"In version 19.0.0.4 and later, some Oracle RAC functions require the use of Oracle UCP."
consider instead
"Some Oracle RAC functions require the use of Oracle UCP which is only available in Open Liberty version 19.0.0.4 and later."
AFAIK, there are no maven coords for RAC.

**Comment by dmuelle** [comment]
Thanks for reviewing @frowe . I've updated the doc per your comments:

https://draft-openlibertyio.mybluemix.net/docs/21.0.0.3/relational-database-connections-JDBC.html#Oraclerac

Let me know if any further edits are needed. If you're satisfied with it, you can add the `Technical reviewed` label to this issue to sign off. Thanks

**Comment by ManasiGandhi** [comment]
@dmuelle #peer review 

The Oracle RAC updates look good! A few edit suggestions for the topic:

- [x] In the second paragraph- "Applications communicate with relational databases to retrieve many different kinds of information", could you change it to, "Applications communicate with relational databases to retrieve different kinds of information, "?

- [x] "Open Liberty recognizes the implementation class names of various data source types for commonly used JDBC drivers. In most cases, you can specify only the location of the JDBC driver.", here do you mean that In most cases, you need to specify only the location of the JDBC driver? Could you rephrase it?
## Data source configuration

- [x] "You can configure any JDBC driver with Open Liberty, which has built-in configuration for many common vendor databases. " Edit suggestion- "You can configure any JDBC driver with Open Liberty, which has a built-in configuration for many common vendor databases. "
- [ ] "If the JDBC driver data source has setter methods with a String or primitive parameter, you can configure these properties by specifying the properties attribute in the dataSource element.", some elaboration on "setter methods" would be helpful.


## Data source types

- [x] Acrolinx is flagging, "If you use the Java Database Connectivity feature 4.2, 4.1, or 4.0 and you are not referencing the default data source, Open Liberty looks for the data source type in the following order" as a long sentence, thought it might be okay as the version numbers are being counted as words.

**Comment by dmuelle** [comment]
Hi @ManasiGandhi , I've updated the file with your suggestions, except for the item about setter methods. Per our slack discussion, setter method is a fundamnetal Java concept that most developers will understand. Let me know if ay further changes are needed

https://draft-openlibertyio.mybluemix.net/docs/latest/relational-database-connections-JDBC.html#Oraclerac

**Comment by Rwalls1** [comment]
# Peer review

## MySQL
The example configuration seems inconsistent when compared to the other configuration examples. In the other examples, the `properties` element specifies the data source that is being configured such as the “Microsoft SQL Server section that includes the example, `properties.microsoft.sqlserver`, which specifies the data source. However, the config example in this section doesn’t include that syntax. Is that discrepancy intentional?


## Oracle RAC

- I think you should another sentence or phrase to explain what nodes are. I understand that it may be a common term, but nodes are not mentioned anywhere else in this topic so I think adding a bit more detail would add more value to this section, especially in regards to understanding the example that is included.
- Is there a reason that this section and the “Oracle UCP” section each include more info than the other “Common data source configuration examples” sections? The other sections only provide minimal info about the databases before introducing the config examples while these Oracle sections provide a deeper introduction to their respective data sources. If there’s not a reason for this discrepancy, I think you should make the sections more parallel by adding more detail to the earlier sections, which would also add more value to the topic. 
- I think the description, "In this example, the host1 and port1 values are for the first Oracle RAC node, and host2 and port2 values are for the second Oracle RAC node.” needs more context. I don’t think it’s fully clear what the distinction is between the “first” and “second” node within the example or why you are highlighting the host and port values in regards to the relevance of those values. Also, since you mentioned that “you can configure failover support, load balancing, or both” in the first paragraph, I think you should specify or describe how those functions are configured in the example, which I assume is with the `FAILOVER` and `LOAD_BALANCE` values. 

## Configuring databases that are unknown to Open Liberty
- I think the title of this section should be changed to match the syntax of the titles of other sections, which don’t use gerunds, so maybe change to something like, “Configuration of unknown databases”
- Typo: Change " specify the mapping of interface class name” to “specify the mapping of the interface class name”

**Comment by dmuelle** [comment]
Thanks for reviewing @Rwalls1. I made the following updates per your review

https://draft-openlibertyio.mybluemix.net/docs/21.0.0.4/relational-database-connections-JDBC.html#Oraclerac

## MySQL
Is that discrepancy intentional?
**yes- i think the reason for the discrepancy is that OL doesn't have a dedicated vendor properties attribute for MySQL as it does for the other datasource examples. Although the config here is essentially the same as the "unknown" datasources, MySQL is a widely-used db that users might be looking for specific info about, and might not realize that this info is included in the "unknown" example.**

## Oracle RAC

- [x]  I think you should another sentence or phrase to explain what nodes are. **nodes are instances of databases that are clustered into a single db, I added a reference to clarify**

- [x] Is there a reason that this section and the “Oracle UCP” section each include more info than the other “Common data source configuration examples” sections? **The other sections are for quick reference of more traditional datasource configuration. The supporting info throughout this topic applies to those examples. The reason there is more info for UCP and RAC is that they are slightly different kinds of options from the other datasources. These types of options are not discussed elsewhere in the topic. UCP and RAC are not a perfect fit in this section, but I think it's better to have them here than on their own page, particularly since they are vendor-specific and OL is open source.** 

- [x]  I think the description, "In this example, the host1 and port1 values are for the first Oracle RAC node, and host2 and port2 values are for the second Oracle RAC node.” needs more context. I don’t think it’s fully clear what the distinction is between the “first” and “second” node within the example or why you are highlighting the host and port values in regards to the relevance of those values. Also, since you mentioned that “you can configure failover support, load balancing, or both” in the first paragraph, I think you should specify or describe how those functions are configured in the example, which I assume is with the `FAILOVER` and `LOAD_BALANCE` values. **updated this section to make it more clear that the example is for an RAC consisting of two nodes and that the config contains example values for each node. Also provided a link to RAC doc as the properties here are Oracle-specific** 

## Configuring databases that are unknown to Open Liberty
-
- [x] I think the title of this section should be changed to match the syntax of the titles of other sections, which don’t use gerunds, so maybe change to something like, “Configuration of unknown databases”



- [x] Typo: Change " specify the mapping of interface class name” to “specify the mapping of the interface class name”

**Comment by Rwalls1** [comment]
@dmuelle Thanks, the updates look good.

**Comment by dmuelle** [comment]
content is on vNext and will publish with 21.0.0.4
